cpp/include/cudf/detail/segmented_reduction_functions.hpp

```diff
@@ -0,0 +1,320 @@
+ *
+ * @param col Input column to compute any
+ * @param offsets Indices to identify segment boundaries
+ * @param output_dtype Data type of return type and typecast elements of input column
+ * @param null_handling If `null_policy::INCLUDE`, all elements in a segment must be valid for the
+ * reduced value to be valid. If `null_policy::EXCLUDE`, the reduced value is valid if any element
+ * in the segment is valid.
+ * @param init Initial value of each any
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned scalar's device memory
+ * @return Column of bool8 for the results of the segments
+ */
+std::unique_ptr<column> segmented_any(
+  column_view const& col,
+  device_span<size_type const> offsets,
+  data_type const output_dtype,
+  null_policy null_handling,
+  std::optional<std::reference_wrapper<scalar const>> init,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Compute if all of the values in the segment are true when typecasted to bool.
+ *
+ * If an input segment is empty, the segment result is null.
+ *
+ * @throw cudf::logic_error if input column type is not convertible to bool.
+ * @throw cudf::logic_error if `output_dtype` is not bool8.
+ *
+ * @param col Input column to compute all
+ * @param offsets Indices to identify segment boundaries
+ * @param output_dtype Data type of return type and typecast elements of input column
+ * @param null_handling If `null_policy::INCLUDE`, all elements in a segment must be valid for the
+ * reduced value to be valid. If `null_policy::EXCLUDE`, the reduced value is valid if any element
+ * in the segment is valid.
+ * @param init Initial value of each all
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned scalar's device memory
+ * @return Column of bool8 for the results of the segments
+ */
+std::unique_ptr<column> segmented_all(
+  column_view const& col,
+  device_span<size_type const> offsets,
+  data_type const output_dtype,
+  null_policy null_handling,
+  std::optional<std::reference_wrapper<scalar const>> init,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Computes mean of elements of segments in the input column
+ *
+ * If input segment is empty, the segment result is null.
+ *
+ * If `null_handling==null_policy::INCLUDE`, all elements in a segment must be valid
+ * for the reduced value to be valid.
+ * If `null_handling==null_policy::EXCLUDE`, the reduced value is valid if any element
+ * in the segment is valid.
+ *
+ * @throw cudf::logic_error if input column type is not arithmetic type
+ * @throw cudf::logic_error if `output_dtype` is not floating point type
+ *
+ * @param col Input column data
```

Comment:
The wording of parameter descriptions for the new reductions differs from the wording of the other segmented reductions (e.g. `@param col Input column data` vs. `@param col Input column to compute all`). Maybe consider standardizing pieces of this?